src/transaction/id.rs

```diff
@@ -362,18 +372,32 @@ mod tests {
                                     } => invert(static_contracts.first_mut().unwrap()),
                                     _ => unreachable!(),
                                 });
-                            }
 
-                            if !storage_slots.is_empty() {
                                 assert_id_ne(&tx, |t| match t {
-                                    Transaction::Create { storage_slots, .. } => {
-                                        invert_storage_slot(storage_slots.first_mut().unwrap())
-                                    }
+                                    Transaction::Create { salt, .. } => invert(salt),
                                     _ => unreachable!(),
                                 });
-                            }
 
-                            assert_id_common_attrs(&tx);
+                                if !static_contracts.is_empty() {
+                                    assert_id_ne(&tx, |t| match t {
+                                        Transaction::Create {
+                                            static_contracts, ..
+                                        } => invert(static_contracts.first_mut().unwrap()),
+                                        _ => unreachable!(),
+                                    });
+                                }
+
+                                if !storage_slots.is_empty() {
```

Comment:
This shouldn't be wrapped inside of `if !static_contracts.is_empty() {` on line 368, the first iteration of the outer loop will always have static_contracts as empty.